### ISSUE TITLE
PropertyDataFetcher cannot cope with nested private classes implementing public interfaces 

### ISSUE THREAD

**Issue by timw6n-thg** [graphql-java/graphql-java#1660]
**Describe the bug**
Looks like the default `graphql.schema.PropertyDataFetcher` is only looking for public getters on classes that are themselves public, and then recurses downwards through the class hierarchy. 

In the case of a private (or package private) class that extends another private class (where the getter is actually coded) and implements a public interface (that defines the getter), the getter is therefore missed altogether. Ends up looking on `Object` and obviously not finding anything.

**To Reproduce**
See minimal failing test case below. Instantiating `Baz` creates a perfectly usable `Foo` in the normal java world, but GraphQL-Java cannot locate the getter for `something`.

```
package com.company.application.upstream;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import graphql.schema.DataFetchingEnvironment;
import graphql.schema.PropertyDataFetcher;

class PropertyDataFetcherInterfacesBug {

  public interface Foo {
    String getSomething();
  }

  private static class Bar implements Foo {
    @Override
    public String getSomething() {
      return "bar";
    }
  }

  // If this were public the test would pass, but it shouldn't need to be.
  private static class Baz extends Bar implements Foo {}

  @Test
  void testPropertyDataFetcherHandlesInterfacesCorrectly() {
    final Foo baz = new Baz();

    final PropertyDataFetcher<String> fetcher = new PropertyDataFetcher<>("something");
    final DataFetchingEnvironment environment = mock(DataFetchingEnvironment.class);
    when(environment.getSource()).thenReturn(baz);

    assertThat(baz.getSomething(), is("bar"));
    assertThat(fetcher.get(environment), is("bar")); // is actually null
  }

}
```

**Comment by andimarek** [comment]
#1737 fixes this: This was actually not supported so far.

**Comment by andimarek** [comment]
closing this issue now, because it will be fixed with #1737
